Source/Scene/ModelExperimental/MetadataPipelineStage.js

```diff
@@ -197,6 +265,26 @@ function addPropertyAttributeProperty(
     MetadataPipelineStage.FUNCTION_ID_INITIALIZE_METADATA_FS,
     [initializationLine]
   );
+
+  // Add lines to set values in the metadataClass struct
+  for (const {
+    specName,
+    shaderName,
+  } of MetadataPipelineStage.METADATACLASS_FIELDS) {
+    const fieldValue = property.classProperty[specName];
+    if (!defined(fieldValue)) {
+      continue;
+    }
+    const fieldLine = `metadataClass.${metadataVariable}.${shaderName} = ${glslType}(${fieldValue});`;
```

Comment:
The template is setting up a type **constructor**. So if `noData == 1`,  and `glslType == 'float'`, we would get this shader line:
```glsl
metadataClass.floatVariable.noData = float(1);
```
I believe the constructors are pretty forgiving: they will do type conversions.

For a unit test, I will have to make a new glTF with floating-point metadata, and then give it integer values for `noData`, `min`, `max`, `default`.

Reply:
ah right, type constructors are a bit more forgiving. Still, would be good to try at least one example if possible since floats are very common property types. You could just add another property to the glTF if needed (it can share the underlying texture/attribute if that makes it easier to generate the test data)

Reply:
Thanks @ptrgags, I set `"default": "1"` for a float property in `SimplePropertyTexture.gltf`, and then updated `MetadataPipelineStageSpec` to make sure we get the following line in the shader:
```glsl
    metadataClass.insulation.defaultValue = float(1);
```
Let me know if this is not what you had in mind.